### ISSUE TITLE
Python bindings

### ISSUE THREAD

**Issue by aschampion** [aschampion/rust-n5#8]
To provide a lower installation barrier, more N5-specific python access to N5 than the excellent [z5py](https://github.com/constantinpape/z5), it would be nice to have python bindings via rust-n5 than supports:

- [ ] dataset creation
- [ ] attribute access
- [ ] numpy array read/write

This would be implemented in a separate, dependent python binding crate (possibly in this repo with a cargo workspace), but is tracked here.

Two main python binding libraries exist for rust ([comparison](https://pyo3.rs/master/rust-cpython.html)):

- [PyO3](https://github.com/PyO3/pyo3)
- [rust-cypython](https://github.com/dgrunwald/rust-cpython)

On first examination I tend to prefer PyO3 (see [this rather nice example](https://github.com/PyO3/pyo3/tree/master/examples/word-count)). [The rust numpy binding library](https://github.com/termoshtt/rust-numpy) currently uses rust-cpython, but plans to migrate (and has an in-progress PR) to PyO3 once it supports stable rust. Will revisit this issue once that has happened.

cc @clbarnes

**Comment by clbarnes** [comment]
Do you envision a numpy-like interface to datasets or would you be more inclined towards going a separate way, like the reference implmentation of `zarr`? Catching enough slicing and broadcasting edge cases to be usefully numpy-compatible is awkward (although in some cases, already implemented in z5).

**Comment by aschampion** [comment]
rust-numpy has migrated to PyO3; their release is blocked on the next release (0.3) of PyO3, but that's nearing. So that blocker has been removed.

**Comment by aschampion** [comment]
This is fulfilled by https://github.com/pattonw/rust-pyn5
